type-o-matic/Program.cs

```diff
@@ -51,12 +51,21 @@ public static int Main (string [] args)
 				return 1;
 			}
 
-	    		if ((options.Framework != null && options.Namespaces.Count > 0) ||
-	    			(options.Framework == null && options.Namespaces.Count == 0)) {
-				Console.WriteLine ("You need to select either input namespaces (to generate swift) or a XamGlue.framework (to generate C#) but not both.");
+			if (options.ToGenerate == null || (!options.ToGenerate.Equals ("swift") && !options.ToGenerate.Equals ("csharp"))) {
+				Console.WriteLine ("You need to specify what type of code you want to generate, either 'swift' or 'csharp'.");
 				return 1;
 			}
-			
+
+			if (options.Namespaces.Count == 0) {
+				Console.WriteLine ("You need to select input namespaces.");
+				return 1;
+			}
+
+			if (options.ToGenerate.Equals ("csharp") && options.Framework == null) {
```

Comment:
Better to use this version of the method: https://docs.microsoft.com/en-us/dotnet/api/system.string.equals?view=netframework-4.8#System_String_Equals_System_String_System_String_System_StringComparison_

Reply:
Fixed!